streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -285,16 +307,18 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
 
       // Distribute the receivers and start them
       logInfo("Starting " + receivers.length + " receivers")
+      terminated = false
       ssc.sparkContext.runJob(tempRDD, ssc.sparkContext.clean(startReceiver))
+      terminated = true
       logInfo("All of the receivers have been terminated")
     }
 
     /** Stops the receivers. */
     private def stopReceivers() {
       // Signal the receivers to stop
-      receiverInfo.values.flatMap { info => Option(info.actor)}
-                         .foreach { _ ! StopReceiver }
-      logInfo("Sent stop signal to all " + receiverInfo.size + " receivers")
+      val receivers = receiverInfo.values.flatMap { info => Option(info.actor) }
+      receivers.foreach { _ ! StopReceiver }
+      logInfo("Sent stop signal to all " + receivers.size + " receivers")
```

Comment:
Might as well use string interpolation here, if you're going to change things anyways:

```scala
logInfo(s"Sent stop signal to all ${receivers.size} receivers")
```